src/java/org/apache/cassandra/config/registry/ConfigurationRegistry.java

```diff
@@ -0,0 +1,371 @@
+        names.put("audit_logging_options_included_keyspaces", "audit_logging_options.included_keyspaces");
+        names.put("audit_logging_options_included_users", "audit_logging_options.included_users");
+        names.put("server_encryption_options_algorithm", "server_encryption_options.algorithm");
+        names.put("server_encryption_options_cipher_suites", "server_encryption_options.cipher_suites");
+        names.put("server_encryption_options_enabled", "server_encryption_options.enabled");
+        names.put("server_encryption_options_internode_encryption", "server_encryption_options.internode_encryption");
+        names.put("server_encryption_options_optional", "server_encryption_options.optional");
+        names.put("transparent_data_encryption_options_chunk_length_kb", "transparent_data_encryption_options.chunk_length_kb");
+        names.put("transparent_data_encryption_options_cipher", "transparent_data_encryption_options.cipher");
+        names.put("transparent_data_encryption_options_enabled", "transparent_data_encryption_options.enabled");
+        names.put("transparent_data_encryption_options_iv_length", "transparent_data_encryption_options.iv_length");
+
+        return names;
+    }
+
+    private static class TransformSet<T, R> extends AbstractSet<R>
+    {
+        private final Set<T> set;
+        private final Function<T, R> transform;
+
+        public TransformSet(Set<T> set, Function<T, R> transform)
+        {
+            this.set = set;
+            this.transform = transform;
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        public Iterator<R> iterator()
+        {
+            Iterator<T> iter = set.iterator();
+            return new AbstractIterator<R>()
+            {
+                @Override
+                protected R computeNext()
+                {
+                    if (!iter.hasNext())
+                        return endOfData();
+
+                    T entry = iter.next();
+                    return transform.apply(entry);
+                }
+            };
+        }
+        /** {@inheritDoc} */
+        @Override
+        public int size()
+        {
+            return set.size();
+        }
+    }
+
+    private static class PropertyChangeListenerWrapper<T> implements PropertyChangeListener<T>
+    {
+        private final PropertyChangeListener<T> listener;
+        private final Class<T> type;
+
+        public PropertyChangeListenerWrapper(PropertyChangeListener<T> listener, Class<T> type)
+        {
+            this.listener = listener;
+            this.type = type;
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        public void onBeforeChange(String name, Object oldValue, Object newValue)
+        {
+            T castedOldValue = type.cast(oldValue);
+            T castedNewValue = type.cast(newValue);
+            listener.onBeforeChange(name, castedOldValue, castedNewValue);
+        }
+    }
```

Comment:
is this needed?  If the user defines the call site as `boolean` the JVM will add the cost for you, so should be able to drop this whole class